frontend/Auth/FinishResetPassword.tsx

```diff
@@ -111,19 +176,48 @@ export function FinishResetPassword(props: RouterProps) {
                 onChange={(evt) =>
                   setFinishResetPasswordFormData({
                     ...finishResetPasswordFormData,
-                    password: evt.target.value,
+                    password: evt.target.value.trim(),
                   })
                 }
                 required
               />
             </FormField>
+            <FormField className="mb-4">
+              <FormFieldLabel
+                className="text-3xl lg:text-xs"
+                htmlFor="passwordCheck"
+              >
+                Re-enter Password
+              </FormFieldLabel>
+              <Input
+                id="passwordCheck"
+                className="text-5xl lg:text-sm"
+                type="password"
+                value={finishResetPasswordFormDataCheck.password}
+                onChange={(evt) =>
+                  setFinishResetPasswordFormDataCheck({
+                    ...finishResetPasswordFormDataCheck,
+                    password: evt.target.value.trim(),
+                  })
+                }
+                required
+              />
+            </FormField>
+            <p className="text-sm text-gray-500">
+              {finishResetPasswordErrors.passwordCheck}
+            </p>
+            <p className="text-sm text-gray-500">
+              {finishResetPasswordErrors.message}
+            </p>
           </div>
 
           <div className="inset-x-0 my-8 bottom-0">
             <Button
               kind={ButtonKind.primary}
+              style={{ opacity: disableSubmit ? "50%" : "100%" }}
```

Comment:
Generally better to use className rather than style when tailwind has classes made for things.

```suggestion
              className={`w-full h-24 lg:h-10 text-3xl lg:text-sm ${disableSubmit ? "opacity-1" : "opacity-.5"}`}
```

Alternatively you can use kremling's `always()` and `maybe()` helpers which can sometimes be nice:

```ts
import { always } from 'kremling';

<Button className={always("w-full h-24 lg:h-10 text-3xl lg:text-sm").toggle("opacity-.5", "opacity-1", disableSubmit)}>
```

Reply:
Interesting! Thank you!